### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by vrezhar**
description isn't updated when barcodes aren't validated

### CODE REVIEW

File: grails-app/services/com/secured/api/DocumentAcceptanceService.groovy [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    {
+        DevCycleLogger.log("acceptProducts() called")
+        Response response = new Response()
+upper:  for(it in productsList)
+        {
+            Products products = Products.findWhere(productCode: it.product_code)
+            if(products)
+            {
+                DevCycleLogger.log("found product with code ${it.product_code}, belonging to company with id ${company.companyId}")
+                boolean contains = document.products?.contains(products)
```
Comment (edasatryan) [inline grails-app/services/com/secured/api/DocumentAcceptanceService.groovy]: > boolean contains = document.products?.contains(products)
need to be Boolean contains = document?.products?.contains(products)

File: grails-app/services/com/secured/api/DocumentAcceptanceService.groovy [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import com.secured.data.Document
+import com.secured.data.Products
+import com.secured.logs.DevCycleLogger
+import grails.gorm.transactions.Transactional
+
+@Transactional
+class DocumentAcceptanceService extends  DocumentService
+{
+
+    protected Map acceptProducts(Document document, Company company, List<ProductCommand> productsList)
```
Comment (edasatryan) [inline grails-app/services/com/secured/api/DocumentAcceptanceService.groovy]: acceptProducts function need to be in ProductsService

File: grails-app/services/com/secured/api/ProductsService.groovy [hunk 3]
```diff
@@ -1,11 +1,127 @@
...
+                }
+                DevCycleLogger.log("barcode with uit code ${barCode.uit_code} and uitu code ${barCode.uitu_code} not validated, nothing updated, exiting update()")
+                return null
+            }
+            products.save()
+            barCode.save(true)
+            DevCycleLogger.log("success, adding to the found product")
+            return products
+        }
+        products.save(true)
```
Comment (edasatryan) [inline grails-app/services/com/secured/api/ProductsService.groovy]: products.save calling much time (Please review and check )

File: grails-app/services/com/secured/api/DocumentAcceptanceService.groovy [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                }
+                DevCycleLogger.log("document with number ${document.documentNumber} not saved(empty products list), exiting accept()")
+                productsList.each {
+                    response_.rejectProduct(it)
+                }
+                return response_.getAsMap()
+            }
+            document.save(true)
+        }
+        DevCycleLogger.log("exiting acceptProducts()")
```
Comment (edasatryan) [inline grails-app/services/com/secured/api/DocumentAcceptanceService.groovy]: The code is not readable. Need to split into separate functions.

File: grails-app/services/com/secured/api/DocumentAcceptanceService.groovy [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+            Response response_ = new Response()
+            response_.reportInvalidInput()
+            cmd.products.each {
+                response_.rejectProduct(it)
+            }
+            return response_.getAsMap()
+        }
+        DevCycleLogger.log("document validated, saving, exiting accept()")
+        return  response
+    }
```
Comment (edasatryan) [inline grails-app/services/com/secured/api/DocumentAcceptanceService.groovy]: Dubliacation of a code

File: grails-app/services/com/secured/api/DocumentService.groovy [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        document.tradeRecipientInn = cmd.trade_recipient_inn
+        return document
+    }
+
+    Document createShipmentDocumentMock(ShipmentDocumentCommand cmd)
+    {
+        Document document = createDocumentMock(cmd)
+        document.requestType = "SHIPMENT"
+        if(cmd.pdf != "" && cmd.pdf != null)
+            document.pdf = cmd.pdf
```
Comment (edasatryan) [inline grails-app/services/com/secured/api/DocumentService.groovy]: please use brackets {}

File: grails-app/services/com/secured/api/DocumentShipmentService.groovy [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        Response response = new Response()
+        for(it in productsList)
+        {
+            Products products = Products.findWhere(productCode:  it.product_code)
+            if(products && company.products?.contains(products))
+            {
+                boolean contains = document.products?.contains(products)
+                DevCycleLogger.log("found product with code ${it.product_code}, belonging to company with id ${company.companyId}")
+                products = super.ship(it)
+                if(!products)
```
Comment (edasatryan) [inline grails-app/services/com/secured/api/DocumentShipmentService.groovy]: can products be null for the 2-nd if? (check and discuss)

File: grails-app/services/com/secured/api/DocumentShipmentService.groovy [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    Map ship(ShipmentDocumentCommand cmd)
+    {
+        DevCycleLogger.log("ship() called")
+        Company company = Company.findWhere(token: cmd.companyToken)
+        if(!company)
+        {
+            DevCycleLogger.log("company with token ${cmd.companyToken} not found, reporting authorization failure, exiting ship()")
+            Response response = new Response()
+            response.rejectCompanyToken()
+            return response.getAsMap()
```
Comment (edasatryan) [inline grails-app/services/com/secured/api/DocumentShipmentService.groovy]: create new function for this part

File: grails-app/services/com/secured/api/DocumentShipmentService.groovy [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+        }
+        if(!cmd.validate())
+        {
+            DevCycleLogger.log("command object not validated, reporting invalid input, exiting ship()")
+            Response response = new Response()
+            response.reportInvalidInput()
+            cmd.products.each {
+                response.rejectProduct(it)
+            }
+            return response.getAsMap()
```
Comment (edasatryan) [inline grails-app/services/com/secured/api/DocumentShipmentService.groovy]: create new function for this part

File: grails-app/services/com/secured/api/DocumentShipmentService.groovy [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+                }
+                DevCycleLogger.log("document with number ${document.documentNumber} not validated, exiting accept()")
+                return response
+            }
+            document.save(true)
+            DevCycleLogger.log("saving document with number ${document.documentNumber}, exiting accept()")
+            return response
+        }
+        DevCycleLogger.log("document with number ${document.documentNumber} not saved(empty products list), exiting accept()")
+        return response
```
Comment (edasatryan) [inline grails-app/services/com/secured/api/DocumentShipmentService.groovy]: please implement a new  function for this part and also use response class as we discussed

File: grails-app/services/com/secured/api/ProductsService.groovy [hunk 11]
```diff
@@ -1,11 +1,127 @@
...
+        }
+        Products products = Products.findWhere(productCode: cmd.product_code)
+        boolean descriptionchanged = false
+        if(cmd.product_description != null && cmd.product_description != "" && cmd.product_description != products.description)
+        {
+            DevCycleLogger.log("updating found product's description")
+            products.description = cmd.product_description
+            descriptionchanged = true
+        }
+        if((cmd.uitu_code != null && cmd.uitu_code != "") || (cmd.uit_code != null && cmd.uit_code != ""))
```
Comment (edasatryan) [inline grails-app/services/com/secured/api/ProductsService.groovy]: ! cmd.uitu_cod <=> cmd.uitu_code != null && cmd.uitu_code != ""

File: grails-app/services/com/secured/api/ProductsService.groovy [hunk 12]
```diff
@@ -1,11 +1,127 @@
...
+        {
+            cmd.errors.fieldErrors.each {
+                DevCycleLogger.log("couldn't validated value ${it.rejectedValue} for field ${it.field}")
+            }
+            DevCycleLogger.log("command object not validated, nothing updated, exiting update()")
+            return null
+        }
+        Products products = Products.findWhere(productCode: cmd.product_code)
+        boolean descriptionchanged = false
+        if(cmd.product_description != null && cmd.product_description != "" && cmd.product_description != products.description)
```
Comment (edasatryan) [inline grails-app/services/com/secured/api/ProductsService.groovy]: !cmd.product_description <=> cmd.product_description != null && cmd.product_description

File: grails-app/services/com/secured/api/DocumentShipmentService.groovy (hunk 7 above)
Comment (vrezhar) [reply grails-app/services/com/secured/api/DocumentShipmentService.groovy]: che productsy erbeq null chi, defaultov datark list a

**Pull request closed without merging**